exchanges/bybit/bybit_test.go

```diff
@@ -85,3502 +74,2995 @@ func TestGetKlines(t *testing.T) {
+	t.Parallel()
+	var feeBuilder = &exchange.FeeBuilder{
+		Amount:              1,
+		FeeType:             exchange.CryptocurrencyTradeFee,
+		Pair:                spotTradablePair,
+		PurchasePrice:       1,
+		FiatCurrency:        currency.USD,
+		BankTransactionType: exchange.WireTransfer,
+	}
+	_, err := b.GetFeeByType(context.Background(), feeBuilder)
+	if err != nil {
+		t.Fatal(err)
+	}
+	feeBuilder.Pair = optionsTradablePair
+	_, err = b.GetFeeByType(context.Background(), feeBuilder)
+	if err != nil {
+		t.Fatal(err)
+	}
+	feeBuilder.Pair = linearTradablePair
+	_, err = b.GetFeeByType(context.Background(), feeBuilder)
+	if err != nil {
+		t.Fatal(err)
 	}
-	if cl != futures.Fortnightly {
-		t.Error("expected fortnightly")
+	feeBuilder.Pair = inverseTradablePair
+	_, err = b.GetFeeByType(context.Background(), feeBuilder)
+	if err != nil {
+		t.Fatal(err)
 	}
+}
 
-	cl, err = getContractLength(time.Hour * 24 * 30 * 3)
-	if !errors.Is(err, nil) {
+func TestSetLeverage(t *testing.T) {
+	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, b, canManipulateRealOrders)
+	ctx := context.Background()
+	b.Verbose = true
+	err := b.SetLeverage(ctx, asset.USDTMarginedFutures, linearTradablePair, margin.Multi, 5, order.Buy)
+	if err != nil {
 		t.Error(err)
 	}
-	if cl != futures.Quarterly {
-		t.Error("expected quarterly")
+
+	err = b.SetLeverage(ctx, asset.CoinMarginedFutures, inverseTradablePair, margin.Isolated, 5, order.UnknownSide)
+	if !errors.Is(err, errOrderSideRequired) {
+		t.Errorf("received '%v', expected '%v'", err, errOrderSideRequired)
 	}
 
-	cl, err = getContractLength(time.Hour * 24 * 30 * 6)
-	if !errors.Is(err, nil) {
+	err = b.SetLeverage(ctx, asset.USDTMarginedFutures, linearTradablePair, margin.Isolated, 5, order.Buy)
+	if err != nil {
 		t.Error(err)
 	}
-	if cl != futures.HalfYearly {
-		t.Error("expected half yearly")
-	}
 
-	cl, err = getContractLength(time.Hour * 24 * 30 * 9)
-	if !errors.Is(err, nil) {
+	err = b.SetLeverage(ctx, asset.CoinMarginedFutures, inverseTradablePair, margin.Isolated, 5, order.Sell)
+	if err != nil {
 		t.Error(err)
 	}
-	if cl != futures.NineMonthly {
-		t.Error("expected nine monthly")
+
+	err = b.SetLeverage(ctx, asset.USDTMarginedFutures, linearTradablePair, margin.Isolated, 5, order.CouldNotBuy)
+	if !errors.Is(err, order.ErrSideIsInvalid) {
+		t.Errorf("received '%v', expected '%v'", err, order.ErrSideIsInvalid)
 	}
 
-	cl, err = getContractLength(time.Hour * 24 * 30 * 12)
-	if !errors.Is(err, nil) {
-		t.Error(err)
+	err = b.SetLeverage(ctx, asset.Spot, inverseTradablePair, margin.Multi, 5, order.UnknownSide)
+	if !errors.Is(err, asset.ErrNotSupported) {
+		t.Errorf("received '%v', expected '%v'", err, asset.ErrNotSupported)
 	}
-	if cl != futures.SemiAnnually {
-		t.Error("expected semi annually")
+}
+
+func TestWsConnecting(t *testing.T) {
+	err := b.WsConnect()
+	if err != nil {
+		t.Error(err)
 	}
+	time.Sleep(time.Second * 25)
```

Comment:
I was wondering why this particular test was taking so long 😆

Reply:
Used to see whether the channel is receiving messages. And forgotten.